host_core/lib/host_core/actors/actor_supervisor.ex

```diff
@@ -45,32 +47,96 @@ defmodule HostCore.Actors.ActorSupervisor do
       with {:ok, {pid, _}} <- VirtualHost.lookup(host_id),
            config <- VirtualHost.config(pid),
            labels <- VirtualHost.labels(pid),
-           {:ok, claims} <- get_claims(bytes, oci),
-           target <- %{
-             publicKey: claims.public_key,
-             issuer: claims.issuer,
-             contractId: nil,
-             linkName: nil
-           },
-           %{permitted: true} <-
-             HostCore.Policy.Manager.evaluate_action(config, labels, source, target, @start_actor),
-           {:ok} <- check_other_oci_already_running(oci, claims.public_key, host_id),
-           pids <- start_actor_instances(claims, bytes, oci, annotations, host_id, count) do
-        Tracer.add_event("Actor(s) Started", [])
-        Tracer.set_status(:ok, "")
-        {:ok, pids}
-      else
-        %{permitted: false, message: message, requestId: request_id} ->
-          Tracer.set_status(:error, "Policy denied starting actor, request: #{request_id}")
-          {:error, "Starting actor denied: #{message}"}
+           {:ok, claims} <- get_claims(bytes, oci) do
+        with target <- %{
```

Comment:
Strong suggestion:

This function has nested a with/else structure that doesn't fit on one "screen" (mostly because the publish functions are split across so many lines

How about splitting this into multiple functions which do one thing, e.g. `validateHost`, `validateAction` which return a common structure, so there can be a single else which emits the failed event and corresponding error?